test/unit/mdc-textfield/mdc-text-field-character-counter-foundation.test.js

```diff
@@ -53,7 +53,9 @@ test('#setContent sets the content of the character counter element', () => {
 });
 
 test('#setContent current length does not exceed character count limit', () => {
-  const {foundation, mockAdapter} = setupTest();
+  const foundation = new MDCTextFieldCharacterCounterFoundation();
+  const adapter = foundation.adapter_;
+  adapter.setContent = td.func('setContent');
```

Comment:
I think I asked this on one of the tab PRs and I'm asking it again here: why are we suddenly using a partially-mocked adapter? What does this improve in code coverage? Is there something else we should be doing to improve it instead?

Reply:
Istanbul was complaining that we're not testing
`new MDCTextFieldCharacterCounterFoundation()` _without_ an argument. I tried adding `/* istanbul ignore next */` comments everywhere, but nothing worked (I'm guessing it's a quirk of transpiling and source maps).

The only thing that made Istanbul happy was testing the no-argument constructor directly like this.

Reply:
Reverted these changes and added a separate test for Istanbul code coverage